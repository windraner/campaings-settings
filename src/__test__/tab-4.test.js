import React from 'react';
import Tab4 from '../components/main-content-section/tab-block/tab-4/tab-4';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const mock = {
  isMediaRights: false,
  hashtagRightsList: ['zxc', 'vbn'],
  termsAndConditions: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.' +
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.' +
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.' +
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.' +
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.' +
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  stateHandler() {},
  switchHandler() {},
};

describe('<Tab4 />', () => {
  it('Check for media rights switcher', () => {
    const wrapper = shallow(<Tab4 {...mock} />);
    expect(wrapper.find('.campaigns-settings__border-separator').length).toBe(0);
    wrapper.setProps({ isMediaRights: true });
    expect(wrapper.find('.campaigns-settings__border-separator').length).toBe(1);
  });
  it('matches the snapshot', () => {
    const tree = shallow(<Tab4 {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
    tree.setProps({ isMediaRights: true });
    expect(toJson(tree)).toMatchSnapshot();
  });
});
