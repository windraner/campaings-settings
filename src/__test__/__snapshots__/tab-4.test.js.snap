// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab4 /> matches the snapshot 1`] = `
<Fragment>
  <TabTitle
    text="Edit Media rights"
  />
  <TabContainer>
    <Switch
      field="isMediaRights"
      isChecked={false}
      switchHandler={[Function]}
      text="Enable media rights framework?"
    />
  </TabContainer>
</Fragment>
`;

exports[`<Tab4 /> matches the snapshot 2`] = `
<Fragment>
  <TabTitle
    text="Edit Media rights"
  />
  <TabContainer>
    <Switch
      field="isMediaRights"
      isChecked={true}
      switchHandler={[Function]}
      text="Enable media rights framework?"
    />
    <hr
      className="campaigns-settings__border-separator"
    />
    <CreatableSelectInput
      field="hashtagRightsList"
      itemsList={
        Array [
          "zxc",
          "vbn",
        ]
      }
      stateHandler={[Function]}
      text="Hashtags for auto-grant rights"
      tooltipText="test message"
    />
    <div
      className="campaigns-settings__divider-mt-32"
    />
    <TextareaInput
      className="campaigns-settings__min-h-320"
      field="termsAndConditions"
      stateHandler={[Function]}
      title="Terms & Conditions for media rights"
      value="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
    />
  </TabContainer>
</Fragment>
`;
